test/gtest/avgpool.hpp

```diff
@@ -0,0 +1,449 @@
+
+        std::vector<size_t> out_grad_dim;
+        if(in_grad_dim.size() == 5)
+        {
+            if(ceil_mode)
+            {
+                OD = std::ceil(static_cast<float>(D - ksize[0] + 2 * padding[0]) / stride[0]) + 1;
+                OH = std::ceil(static_cast<float>(H - ksize[1] + 2 * padding[1]) / stride[1]) + 1;
+                OW = std::ceil(static_cast<float>(W - ksize[2] + 2 * padding[2]) / stride[2]) + 1;
+            }
+            else
+            {
+                OD = std::floor(static_cast<float>(D - ksize[0] + 2 * padding[0]) / stride[0]) + 1;
+                OH = std::floor(static_cast<float>(H - ksize[1] + 2 * padding[1]) / stride[1]) + 1;
+                OW = std::floor(static_cast<float>(W - ksize[2] + 2 * padding[2]) / stride[2]) + 1;
+            }
+            out_grad_dim = {N, C, OD, OH, OW};
+        }
+        else
+        {
+            if(ceil_mode)
+            {
+                OH = std::ceil(static_cast<float>(H - ksize[0] + 2 * padding[0]) / stride[0]) + 1;
+                OW = std::ceil(static_cast<float>(W - ksize[1] + 2 * padding[1]) / stride[1]) + 1;
+            }
+            else
+            {
+                OH = std::floor(static_cast<float>(H - ksize[0] + 2 * padding[0]) / stride[0]) + 1;
+                OW = std::floor(static_cast<float>(W - ksize[1] + 2 * padding[1]) / stride[1]) + 1;
+            }
+            out_grad_dim = {N, C, OH, OW};
+        }
+
+        auto gen_output_grad_value = [](auto...) {
+            return prng::gen_A_to_B<T>(static_cast<T>(-10.0f), static_cast<T>(10.0f));
+        };
+        auto out_grad_strides = avgpool_config.ComputeStrides(out_grad_dim);
+        output_grad = tensor<T>{out_grad_dim, out_grad_strides}.generate(gen_output_grad_value);
+
+        input_grad = tensor<T>{in_grad_dim};
+        std::fill(input_grad.begin(), input_grad.end(), std::numeric_limits<T>::quiet_NaN());
+
+        ref_input_grad = tensor<T>{in_grad_dim};
+        std::fill(
+            ref_input_grad.begin(), ref_input_grad.end(), std::numeric_limits<T>::quiet_NaN());
+
+        output_grad_dev = handle.Write(output_grad.data);
+        input_grad_dev  = handle.Write(input_grad.data);
+    }
+
+    void RunTest()
+    {
+        auto&& handle = get_handle();
+
+        miopenStatus_t status;
+
+        auto dims = input_grad.desc.GetNumDims();
+        if(dims == 4)
+        {
+            cpu_avgpool_backward_2d(output_grad,
+                                    ref_input_grad,
+                                    static_cast<long>(N),
+                                    static_cast<long>(C),
+                                    static_cast<long>(H),
+                                    static_cast<long>(W),
+                                    static_cast<long>(OH),
+                                    static_cast<long>(OW),
+                                    ksize,
+                                    stride,
+                                    padding,
```

Comment:
these are tensor<int64_t> but cpu_avgpool_backward_2d received tensor<long>. This is related to above comment. Also check `cpu_avgpool_backward_3d`

Reply:
resolved